R/tar-terra-vect.R

```diff
@@ -67,89 +71,75 @@
-              )
-              file.rename(paste0(path, ".shz"), path)
-          },
-          marshal = function(object) terra::wrap(object),
-          unmarshal = function(object) terra::unwrap(object)
-      )
-  } else {
-      format <- create_format_terra_vect(filetype, options = gdal, ...)
-  }
-
-  targets::tar_target_raw(
-    name = name,
-    command = command,
-    pattern = pattern,
-    packages = packages,
-    library = library,
-    format = format,
-    repository = repository,
-    iteration = iteration,
-    error = error,
-    memory = memory,
-    garbage_collection = garbage_collection,
-    deployment = deployment,
-    priority = priority,
-    resources = resources,
-    storage = storage,
-    retrieval = retrieval,
-    cue = cue
-  )
+                           command,
+                           pattern = NULL,
+                           filetype = NULL,
+                           gdal = NULL,
+                           ...,
+                           packages = targets::tar_option_get("packages"),
+                           tidy_eval = targets::tar_option_get("tidy_eval"),
+                           library = targets::tar_option_get("library"),
+                           repository = targets::tar_option_get("repository"),
+                           iteration = targets::tar_option_get("iteration"),
+                           error = targets::tar_option_get("error"),
+                           memory = targets::tar_option_get("memory"),
+                           garbage_collection = targets::tar_option_get("garbage_collection"),
+                           deployment = targets::tar_option_get("deployment"),
+                           priority = targets::tar_option_get("priority"),
+                           resources = targets::tar_option_get("resources"),
+                           storage = targets::tar_option_get("storage"),
+                           retrieval = targets::tar_option_get("retrieval"),
+                           cue = targets::tar_option_get("cue")) {
+    name <- targets::tar_deparse_language(substitute(name))
+
+    envir <- targets::tar_option_get("envir")
+
+    command <- targets::tar_tidy_eval(
+        expr = as.expression(substitute(command)),
+        envir = envir,
+        tidy_eval = tidy_eval
+    )
+    pattern <- targets::tar_tidy_eval(
+        expr = as.expression(substitute(pattern)),
+        envir = envir,
+        tidy_eval = tidy_eval
+    )
+
+    # TODO: pull defaults from geotargets package options
+    if (is.null(filetype)) {
+        filetype <- "GeoJSON"
+    }
+
+    if (is.null(gdal)) {
+        gdal <- "ENCODING=UTF-8"
+    }
+
+    if (filetype == "ESRI Shapefile") {
+        #special handling of ESRI shapefiles because the output is a dir of multiple files.
+        format <- create_format_terra_vect_shz(options = gdal, ...)
+    } else {
+        format <- create_format_terra_vect(filetype, options = gdal, ...)
+    }
```

Comment:
I would say put this into an ifelse, like so:

```suggestion
format <- ifelse(
  test = filetype == "ESRI Shapefile",
  #special handling of ESRI shapefiles because the output is a dir of multiple files.
  yes = create_format_terra_vect_shz(options = gdal, ...),
  no =  create_format_terra_vect(filetype, options = gdal, ...)
)
```

If we want to use more than 2, then I think `switch()` is worthwhile

Reply:
I haven't done a thorough exploration of what other drivers might need special handling like this, so this is fine with me